build.gradle

```diff
@@ -99,3 +100,67 @@ wpi.java.configureTestTasks(test)
 tasks.withType(JavaCompile) {
     options.compilerArgs.add '-XDstringConcat=inline'
 }
+
+def getRobotVersionData() { 
+    def robotVersion = new StringBuilder()     
+    robotVersion.append("Commit: @"+"git rev-parse --short HEAD".execute().text.replace('\n', '')+" by "+"git config user.name".execute().text+"\n")
+    robotVersion.append("Branch: ")
+    robotVersion.append('git branch --show-current'.execute().text)
+    robotVersion.append("Uncommitted Changes: ")
+    robotVersion.append('git diff --shortstat'.execute().text)
+    robotVersion.append("--------------------\n")
+    robotVersion.append('git status -s'.execute().text)
+    return robotVersion.toString()
+}
+
+def getMinifiedRobotVersionData() {
+    def miniData = new StringBuilder()
+    miniData.append("git config user.name".execute().text + "@")
+    miniData.append("git branch --show-current".execute().text)
+    miniData.append(" | "+"git rev-parse --short HEAD".execute().text)
+    miniData.append(" | "+("git status".execute().text.length() > 0 ? "*" : ""))
+    return miniData.toString().replace('\n', '')
+}
+
+def writeVersionToFile = { ->
+    if("git config user.name".execute().text.length() < 1) {
+        println "Git username not configured, skipping robot version file writing. We are probably on the build server."
+        return;
+    }
+    
+    def versionDeployDir = new File("./src/main/deploy/version");
+
+    if(!versionDeployDir.exists()) {
+        if(! versionDeployDir.mkdir()) {
+            println "Version deploy directory could not be created."
+        }
+    }
+
+    def versionFile = new File(versionDeployDir, ".robotVersion")
+    def versionFileMini = new File(versionDeployDir, ".robotVersionMini")
+    if(versionFile.exists() || versionFileMini.exists()){
+        versionFile.delete()
```

Comment:
Can you look up if this `delete()` fails if the file doesnt exist? If it does we could get into a weird state possibly if one exists where we can't repair the state without going into the roborio via terminal. If it doesnt fail if the file doesn't exist, this is fine

Reply:
It does fail, I will adjust that logic.